src/ctelnet.cpp

```diff
@@ -813,31 +813,38 @@ void cTelnet::sendTelnetOption(char type, char option)
 }
 
 
-void cTelnet::replyFinished(QNetworkReply* reply)
+void cTelnet::slot_replyFinished(QNetworkReply* reply)
 {
     mpProgressDialog->close();
-    packageDownloadReply->deleteLater();
 
-    // don't process if download was aborted
-    if (!reply->isFinished() || reply->error() != QNetworkReply::NoError) {
-        return;
-    }
+    if (reply != mpPackageDownloadReply) {
+        qWarning().nospace().noquote() << "cTelnet::slot_replyFinished(QNetworkReply*) ERROR - received argument was not the expected stored pointer.";
```

Comment:
I suggest adding more context for this - because the error will be vague otherwise:

```suggestion
        qWarning().nospace().noquote() << "cTelnet::slot_replyFinished(QNetworkReply*) ERROR - download finished, but it wasn't the one we are expecting";
```